### PR TITLE
feat: Add duplicate site fields detection job with comprehensive test coverage

### PR DESCRIPTION
**Pull request by Baalmart**
## Description
feat: Add duplicate site fields detection job with comprehensive test coverage

## Changes Made

- [x] Implemented new job script to detect and report sites with duplicate field values
- [x] Added comprehensive unit tests with edge case coverage
- [x] Fields monitored: `name`, `search_name`, `description`
- [x] Configurable field list for easy maintenance
- [x] Scheduled job running every 2 hours

## Testing

- [x] Tested locally
- [ ] Tested against staging environment
- [ ] Relevant tests passed: [List test names]

## Affected Services

- [ ] Which services were modified:
  - [x] Device Registry

     
## API Documentation Updated?

- [ ] Yes, API documentation was updated
- [x] No, API documentation does not need updating

## Summary by CodeRabbit

- **New Features**
	- Introduced a scheduled job to check for duplicate fields in site records, enhancing data integrity.
  
- **Tests**
	- Added a comprehensive suite of unit tests for the duplicate checking functionality, ensuring robustness and accuracy.

- **Chores**
	- Updated server configuration to include the new duplicate checking job, maintaining existing functionality.

### CODE REVIEW

File: src/device-registry/bin/jobs/test/ut_check-duplicate-site-fields-job.js [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+
+      await checkDuplicateSiteFields();
+
+      expect(
+        loggerStub.error.calledWith(
+          "🐛 Error checking duplicate site fields: Database operation timeout"
+        )
+      ).to.be.true;
+    });
+  });
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply src/device-registry/bin/jobs/test/ut_check-duplicate-site-fields-job.js:306]: create a new Github issue based on this suggestion

File: src/device-registry/bin/jobs/test/ut_check-duplicate-site-fields-job.js [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+
+      findStub.resolves(mockSites);
+
+      await checkDuplicateSiteFields();
+
+      // Should not detect null values as duplicates
+      expect(logTextStub.calledWith("✅ No duplicate field values found")).to.be
+        .true;
+    });
+  });
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply src/device-registry/bin/jobs/test/ut_check-duplicate-site-fields-job.js:424]: create a new Github issue based on this suggestion
